### PR TITLE
Receptive field position

### PR DESCRIPTION
**Pull request by dimitriskor**
Added the position (x, y) of the receptive field in the Cartesian plane as optimizable  parameters.

### CODE REVIEW

File: norse/torch/functional/receptive_field.py [hunk 1]
```diff
@@ -7,20 +7,26 @@
...
+      x (torch.Tensor): The receptive's field center position in x-axis
+      y (torch.Tensor): The receptive's field center position in y-axis
       domain (int): The domain of the kernel. Defaults to 8 (sampling -8 to 8).
     """
     ci = torch.linalg.inv(c)
     cd = torch.linalg.det(c)
     fraction = 1 / (2 * torch.pi * torch.sqrt(cd))
     a = torch.linspace(-domain, domain, size)
     xs, ys = torch.meshgrid(a, a, indexing="xy")
+    xs = xs - x
```
Comment (Jegp) [inline norse/torch/functional/receptive_field.py:28]: This effectively means that a positive x leads to a left-shift. Is that on purpose?
According to the docs, the I would assume a positive x shifts the RF to the right.

File: norse/torch/module/receptive_field.py [hunk 2]
```diff
@@ -157,16 +160,22 @@ def __init__(
         scales: torch.Tensor,
         angles: torch.Tensor,
         ratios: torch.Tensor,
+        x: torch.Tensor,
```
Comment (Jegp) [inline norse/torch/module/receptive_field.py:163]: Would it make sense to default the x, y coordinates to 0? I assume most people want centered RFs. Or?

File: norse/torch/module/receptive_field.py [hunk 3]
```diff
@@ -229,15 +247,19 @@ def __init__(
...
```
Comment (Jegp) [inline norse/torch/module/receptive_field.py:250]: See previous comment.

**Review by Jegp — COMMENTED** [review]
Very nice work here @dimitriskor! You've added a robust change with great tests to avoid regressions. I would perhaps suggest to have a default parameter value for the x,y coordinates *in the class-based API*. What do you think?

File: norse/torch/functional/receptive_field.py (hunk 1 above)
Comment (dimitriskor) [reply norse/torch/functional/receptive_field.py:28]: A positive x leads to a left-shift of the domain, which results in the kernel shifting to the right in the transformed domain. 
A value x<sub>0</sub> performs the transformation g(x) --> g(x-x<sub>0</sub>)

File: norse/torch/module/receptive_field.py (hunk 2 above)
Comment (dimitriskor) [reply norse/torch/module/receptive_field.py:163]: I agree with your comment. I will add the default to 0

File: norse/torch/functional/receptive_field.py (hunk 1 above)
Comment (Jegp) [reply norse/torch/functional/receptive_field.py:28]: Ah, I see. That actually makes sense, thank you!

**Review by Jegp — APPROVED** [review]
Nice work and follow-up @dimitriskor. LGTM!

**Pull request merged**